app/models/post_mover.rb

```diff
@@ -66,6 +66,11 @@ def move_posts_to(topic)
     Guardian.new(user).ensure_can_see! topic
     @destination_topic = topic
 
+    original_topic_posts_count = @original_topic.posts
+      .where("post_type = ? or (post_type = ? and action_code != 'split_topic')", Post.types[:regular], Post.types[:whisper])
```

Comment:
I am curious about the second part of the condition, where you skip split_topic small actions. I understand you do this to skip the small action that is generated right before this method is called. I think there is an edge case here if you move some posts two times in a row. First time you do not select all posts, do the action and a small action post is created. Then you do it again, without selecting the small action created previously. The system will think you selected all posts, but no, because you did not select the small action from the first move.

I might be wrong and even if I am not, I do not think it makes sense fixing it, unless you find something more serious.

Reply:
> I am curious about the second part of the condition

Yeah, it's a pretty tricky place :), I myself spent some time today trying to remember what does it mean. And it was me who [introduced](https://github.com/discourse/discourse/pull/13200) this condition a month ago :), I think it's worth adding the comment here.

I've added a comment. Could you please take a look at it and confirm if it makes sense?

And here is the PR that introduced a change that we support with this condition - https://github.com/discourse/discourse/pull/7085.

Speaking of this:

> I understand you do this to skip the small action that is generated right before this method is called.

We shouldn't have such problems here no matter when a small action was added, we just always ignore small actions when moving posts.